File: app/src/main/java/com/photoeditor/sketch/photo/maker/editor/sketchphotoeditor/pencilsketching/Ragnarok/PixelateFilter.java
package com.photoeditor.sketch.photo.maker.editor.sketchphotoeditor.pencilsketching.Ragnarok;

import android.graphics.Bitmap;

class PixelateFilter {
	static {
		System.loadLibrary("AndroidImageFilter");
	}
	
	static Bitmap changeToPixelate(Bitmap bitmap, int pixelSize) {
		int width = bitmap.getWidth();
		int height = bitmap.getHeight();
		
		int[] pixels = new int[width * height];
		bitmap.getPixels(pixels, 0, width, 0, 0, width, height);
		
		int[] returnPixels = NativeFilterFunc.pxelateFilter(pixels, width, height, pixelSize);

		return Bitmap.createBitmap(returnPixels, width, height, Bitmap.Config.ARGB_8888);
		
	}
}

File: app/src/main/java/com/sketch/ragnarok/PixelateFilter.kt
package com.sketch.ragnarok

import android.graphics.Bitmap

internal object PixelateFilter {
    init {
        System.loadLibrary("AndroidImageFilter")
    }

    fun changeToPixelate(bitmap: Bitmap, pixelSize: Int): Bitmap {
        val width = bitmap.width
        val height = bitmap.height
        val pixels = IntArray(width * height)
        bitmap.getPixels(pixels, 0, width, 0, 0, width, height)
        val returnPixels = NativeFilterFunc.pxelateFilter(pixels, width, height, pixelSize)
        return Bitmap.createBitmap(returnPixels, width, height, Bitmap.Config.ARGB_8888)
    }
}